### PR TITLE
[Kernel] Add Kernel component

### PR DESCRIPTION
**Pull request by alquerci**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #29275, #27479
| License       | MIT
| Doc PR        | TODO

See #29275 for all details.

To help code review there are one commit per action.

Todo
----

- [ ] Release specifications, see questions on https://github.com/symfony/symfony/issues/29275#issuecomment-473461324
- [x] Create the PR to add the kernel component (@alquerci)
    - [x] List test cases
    - [x] Implements tests cases (@alquerci)
    - [x] Write how to migrate on `UPGRADE-6.0.md` (@alquerci)
- [ ] Create the PR for update the documentation

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/composer.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
         "symfony/contracts": "^1.0.2",
         "symfony/event-dispatcher": "^4.3",
         "symfony/http-foundation": "^4.1.1",
+        "instinct/kernel": "dev-symfony-master",
```
Comment (alquerci) [inline src/Symfony/Component/HttpKernel/composer.json]: This is temporary until the component is available on Packagist.

**Comment by alquerci** [comment]
The Travis failure is due to `symfony/messenger` outdated on Packagist `bdf294f` instead of `326b694`.

File: src/Symfony/Component/HttpKernel/Tests/Bundle/BundleTest.php [hunk 2]
```diff
@@ -13,9 +13,13 @@
 
 use PHPUnit\Framework\TestCase;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
+use Symfony\Component\HttpKernel\Bundle\BundleInterface;
 use Symfony\Component\HttpKernel\Tests\Fixtures\ExtensionNotValidBundle\ExtensionNotValidBundle;
 use Symfony\Component\HttpKernel\Tests\Fixtures\ExtensionPresentBundle\ExtensionPresentBundle;
 
+/**
+ * TODO Trigger class deprecation on version 5.1.
```
Comment (alquerci) [inline src/Symfony/Component/HttpKernel/Tests/Bundle/BundleTest.php]: Tests are not covered by backward compatibility promise.

Change the text with:
`Remove on version 6.0. While now it ensure BC layer.`

**Comment by alquerci** [comment]
@Korbeil @DanieleGBX I heard that you want to help in order to speed-up the merge of the new Kernel component. Any help are welcome from anyone else wants to help.

This PR provides all required code modifications, but we still needs to upgrade the documentation and to write the `UPGRADE_6.0.md` file. And for sure code review are welcome.

You can see the full to do list on this PR description.

**Comment by alquerci** [comment]
As it left only to upgrade the documentation then removed `[WIP]`.

cc @stof @nicolas-grekas

**Comment by Simperfit** [comment]
what's the ETA of this ?

**Comment by alquerci** [comment]
@Simperfit As you can see the Dev is done except for the documentation PR that I am not able to do.

I am just waiting for the green fire in order to rebase on the latest `master` version (currently based on 682855f).

@nicolas-grekas @dunglas What is the ETA for the code review? What is the priority of this?

**Comment by fabpot** [comment]
I've just re-read this issue and some other related to the same. I don't think this is something we will ever do. Lately, we have decoupled things more. And Flex did part of the job. But, having big BC breaks is out of the equation with the maturity of the framework. We might do some small steps in this direction, but no big bang. I'm closing for these reasons.

**Pull request closed without merging**